### PR TITLE
Use pimpl avc extractor

### PR DESCRIPTION
**Pull request by mortenvp**

### CODE REVIEW

**Comment by mortenvp** [comment]
@jpihl Please take a look

File: src/petro/extractor/avc_sample_extractor.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        assert(m_impl);
+        return m_impl->sps_size();
+    }
+
+    uint8_t avc_sample_extractor::nalu_length_size() const
+    {
+        assert(m_impl);
+        return m_impl->nalu_length_size();
+    }
```
Comment (jpihl) [inline src/petro/extractor/avc_sample_extractor.cpp]: too much indent here: https://github.com/steinwurf/petro/blob/4450c3f58c77e3ec3940b9d2ed2a063276c303a6/src/petro/extractor/avc_sample_extractor.cpp#L18-L141

File: src/petro/extractor/avc_sample_extractor.cpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        parser_layer<
+        memory_mapped_file_layer>>>>
+    { };
+
+    avc_sample_extractor::avc_sample_extractor() :
+        m_impl(new avc_sample_extractor::impl())
+    { }
+
+    avc_sample_extractor::~avc_sample_extractor()
+    { }
```
Comment (jpihl) [inline src/petro/extractor/avc_sample_extractor.cpp]: is it necessary to have this empty destructor?

File: src/petro/wscript_build [hunk 3]
```diff
@@ -7,3 +7,19 @@ bld.stlib(
...
+
+if '-fPIC' in bld.env.CXXFLAGS:
+
+    bld.shlib(
+        features='cxx',
+        source=bld.path.ant_glob('**/*.cpp'),
+        target='petro_shared',
+        use=['boost_includes', 'boost_iostreams'],
+        export_includes=['..']
+    )
```
Comment (jpihl) [inline src/petro/wscript_build]: It's fine for now, but I think we should find a cleaner solution for this at some point.

File: src/petro/extractor/avc_sample_extractor.cpp (hunk 2 above)
Comment (mortenvp) [inline src/petro/extractor/avc_sample_extractor.cpp]: added comment with explanation

File: src/petro/extractor/avc_sample_extractor.cpp (hunk 1 above)
Comment (mortenvp) [inline src/petro/extractor/avc_sample_extractor.cpp]: fixed

File: src/petro/wscript_build (hunk 3 above)
Comment (mortenvp) [inline src/petro/wscript_build]: good

**Comment by jpihl** [comment]
looks good

**Pull request merged**